src/impl/tpls/KokkosSparse_spmv_tpl_spec_decl.hpp

```diff
@@ -74,21 +90,24 @@ namespace Impl {
 #if defined(CUSPARSE_VERSION) && (10300 <= CUSPARSE_VERSION)
 
     /* Check that cusparse can handle the types of the input Kokkos::CrsMatrix */
-    cusparseIndexType_t myCusparseIndexType;
-    if(std::is_same<offset_type, int>::value)     {myCusparseIndexType = CUSPARSE_INDEX_32I;}
-    if(std::is_same<offset_type, int64_t>::value) {myCusparseIndexType = CUSPARSE_INDEX_64I;}
+    cusparseIndexType_t myCusparseOffsetType;
+    if(std::is_same<offset_type, int>::value)     {myCusparseOffsetType = CUSPARSE_INDEX_32I;}
+    if(std::is_same<offset_type, int64_t>::value) {myCusparseOffsetType = CUSPARSE_INDEX_64I;}
+    cusparseIndexType_t myCusparseEntryType;
+    if(std::is_same<entry_type, int>::value)      {myCusparseEntryType = CUSPARSE_INDEX_32I;}
+    if(std::is_same<entry_type, int64_t>::value)  {myCusparseEntryType = CUSPARSE_INDEX_64I;}
     cudaDataType myCudaDataType;
-    if(std::is_same<value_type, float>::value)  {myCudaDataType = CUDA_R_32F;}
-    if(std::is_same<value_type, double>::value) {myCudaDataType = CUDA_R_64F;}
+    if(std::is_same<value_type, float>::value)    {myCudaDataType = CUDA_R_32F;}
+    if(std::is_same<value_type, double>::value)   {myCudaDataType = CUDA_R_64F;}
```

Comment:
Any reason why ``CUDA_C_32F`` and ``CUDA_C_64F`` aren't supported here? It's ok if those use the fallback SPMV for now, but if it will be done later, we should make a note/issue for it

Reply:
I guess I answered that in your comment below.